refactor: reorganize evaluation scripts structure

chani0343 opened the pull request:
- Move evaluator.py to src/evaluation/base/
- Move metrics files to src/evaluation/metrics/
- Move result files to src/evaluation/results/
- Update all imports to reflect new structure
- Rename complete_hierarchy_results.json to hierarchical_evaluation_results.json
- Update output paths in evaluation scripts

Resolves #124

chani0343 committed:
refactor: reorganize evaluation scripts structure

- Move evaluator.py to src/evaluation/base/
- Move metrics files to src/evaluation/metrics/
- Move result files to src/evaluation/results/
- Update all imports to reflect new structure
- Rename complete_hierarchy_results.json to hierarchical_evaluation_results.json
- Update output paths in evaluation scripts

Resolves #124

lyuzinmaxim commented on src/evaluation/run_evaluation.py:
```diff
@@ -4,10 +4,10 @@
-from src.evaluation.evaluator import EvaluationConfig, ModelEvaluator
+from src.evaluation.base.evaluator import EvaluationConfig, ModelEvaluator
 from src.inference import MobileNetInference, ResNetInference
 from src.onnx_model import OnnxClassifierInferenceBase as OnnxModel
```
could you please import as is, not `as OnnxModel`? Just use `OnnxClassifierInferenceBase`

lyuzinmaxim commented on src/evaluation/results/evaluation_results.json:
Did you make sure that the results remain the same?

chani0343 replied:
What do you mean? Are the results the same? I just arranged the folders, I didn't touch the logic at all.
And we haven't yet integrated into the project that the model will predict better with a black background, so there shouldn't be a change, and indeed, when we do that, we'll run the RUN_EVALOTION again and see the best results.

lyuzinmaxim replied:
Please note that it's a part of the ticket to run the code before your changes, and run after your changes, and at the end to compare the results. The goal behind it is to ensure we didn't break anything - but not to improve results etc. Please do that

lyuzinmaxim commented on src/evaluation/results/hierarchical_evaluation_results.json:
Did you make sure that the results remain the same?

chani0343 committed:
fix: use full class name instead of alias for OnnxClassifierInferenceBase

- Remove 'as OnnxModel' alias as requested in code review
- Use OnnxClassifierInferenceBase directly for better readability

Addresses reviewer feedback

chani0343 committed:
fix: update imports and class names as per code review

chani0343 committed:
Add evaluation results after directory refactoring

- Updated evaluation_results.json with PyTorch model performance
- ResNet18: 26.2% accuracy, 84.9ms latency
- MobileNet: 27.2% accuracy, 55.1ms latency
- Results generated after dataset/preparation and src/evaluation refactoring

chani0343 committed:
Resolve merge conflicts
